test/shooting/basic_problems_tests.jl

```diff
@@ -35,12 +38,13 @@
         @test SciMLBase.successful_retcode(sol)
         @test norm(sol.resid, Inf) < 1e-8
 
+        JET_SKIP[i] && continue
         @test_opt target_modules=(BoundaryValueDiffEq,) solve(
             bvp1, solver; abstol = 1e-8, reltol = 1e-8,
-            odesolve_kwargs = (; abstol = 1e-6, reltol = 1e-3))
+            odesolve_kwargs = (; abstol = 1e-6, reltol = 1e-3)) broken=JET_BROKEN[i]
```

Comment:
what's wrong with the JET tests?

Reply:
JET skipping here was introduced by @avik-pal, since most of the JET tests are marked as broken, I am thinking maybe we can get rid of them? If so, the only thing blocking this PR is #191

Reply:
@avik-pal What's left to deal with is the `MultipleShooting` for Ray tracing oop example error about `f` being non-automatic-differentiable(iip tests working fine), which is weird, I remember before the ADTypes upgrading they were working fine. Any idea how to fix this?